google/cloud/firestore_v1/query.py

```diff
@@ -244,15 +268,27 @@ def stream(
             :class:`~google.cloud.firestore_v1.document.DocumentSnapshot`:
             The next document that fulfills the query.
         """
-        request, expected_prefix, kwargs = self._prep_stream(
+        response_iterator, expected_prefix = self._get_stream_iterator(
             transaction, retry, timeout,
         )
 
-        response_iterator = self._client._firestore_api.run_query(
-            request=request, metadata=self._client._rpc_metadata, **kwargs,
-        )
+        last_snapshot = None
+
+        while True:
+            try:
+                response = next(response_iterator)
```

Comment:
Could we inject this retry into `__iter__` somehow? this seems more generally useful to me.

Reply:
Because the underlying API call is a `unary_stream`, the `response_iterator` is an instance of `google.api_core.grpc_helpers._StreamingResponseIterator`: I guess we could modify its `__next__` to do something like this (it is already tweaked to cache the first result, for instance).

Reply:
I don't see how we get the applicable `retry` object inside there -- the `_wrapped` attribute isn't our `_GapicCallable`, but something altogether different.